File: frontend/types/analytics.ts
export interface WeekLoadDay {
  day: number;
  load_score: number;
  lessons_count: number;
  hard_subjects: string[];
  warning: string | null;
}

export interface WeekLoadAnalysis {
  days: WeekLoadDay[];
}

export interface MoodStats {
  easy_count: number;
  normal_count: number;
  hard_count: number;
}

export interface WarningsResponse {
  warnings: string[];
}

File: backend/app/schemas/analytics.py
from __future__ import annotations

from pydantic import BaseModel, Field


class DayLoadAnalysis(BaseModel):
    day: int = Field(ge=0, le=6)
    load_score: int = Field(ge=0)
    lessons_count: int = Field(ge=0)
    hard_subjects: list[str]
    warning: str | None = None


class WeekLoadAnalysis(BaseModel):
    days: list[DayLoadAnalysis]


class LoadWarningsResponse(BaseModel):
    warnings: list[str]
